ethcore/src/snapshot/service.rs

```diff
@@ -497,7 +497,7 @@ impl Service {
 							match is_done {
 								true => {
 									try!(db.flush().map_err(::util::UtilError::SimpleString));
-									self.finalize_restoration(&mut *restoration)
+									return self.finalize_restoration(&mut *restoration);
```

Comment:
explicit `drop(db)` might be necessary for full correctness, I believe `finalize_restoration` expects no db handles to be live but the cleanup for locals in `feed_chunk` wouldn't be done until after it completes.